sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1054 @@
+     * @return negative errno on error
+     */
+    int (*confirm_cca)(ieee802154_dev_t *dev);
+
+    /**
+     * @brief Get a cap from the radio
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param cap cap to be checked
+     *
+     * @return true if the radio supports the cap
+     * @return false otherwise
+     */
+    bool (*get_cap)(ieee802154_dev_t *dev, ieee802154_rf_caps_t cap);
+
+    /**
+     * @brief Set the threshold for the Energy Detection (first mode of CCA)
+     *
+     * @pre the device is on
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] threshold the threshold in dBm.
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_cca_threshold)(ieee802154_dev_t *dev, int8_t threshold);
+
+    /**
+     * @brief Set CCA mode
+     *
+     * All radios MUST at least implement the first CCA mode (ED Threshold).
+     *
+     * @pre the device is on
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] mode the CCA mode
+     *
+     * @return 0 on success
+     * @return -ENOTSUP if the mode is not supported
+     * @return negative errno on error
+     */
+    int (*set_cca_mode)(ieee802154_dev_t *dev, ieee802154_cca_mode_t mode);
+
+    /**
+     * @brief Set IEEE802.15.4 PHY configuration (channel, TX power)
+     *
+     * This function SHOULD NOT validate the PHY configurations unless
+     * it's specific to the device. The upper layer is responsible of all kind
+     * of validations.
+     * In case a configuration is not valid (e.g parameters out of range), this
+     * function should return -EINVAL
+     *
+     * @pre the device is on
+     * @post the upper layer calls @ref
+     * ieee802154_radio_ops::request_set_trx_state immediately after calling
+     * this function.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] conf the PHY configuration
+     *
+     * @return 0 on success
+     * @return -EINVAL if the configuration is not valid for the device.
+     * @return negative errno on error
+     */
+    int (*config_phy)(ieee802154_dev_t *dev, ieee802154_phy_conf_t *conf);
+
+    /**
+     * @brief Set IEEE802.15.4 addresses in hardware address filter
+     *
+     * @pre the device is on
+     *
+     * @note this function pointer can be NULL if the device doesn't support
+     *       hardware address filtering.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] short_addr the IEEE802.15.4 short address
+     * @param[in] ext_addr the IEEE802.15.4 extended address (Network Byte Order)
+     * @param[in] pan_id the IEEE802.15.4 PAN ID
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_hw_addr_filter)(ieee802154_dev_t *dev, network_uint16_t *short_addr,
+                              eui64_t *ext_addr, uint16_t pan_id);
```

Comment:
If I only want to change, e.g. the `pan_id`, can `short_addr` and `ext_addr` be `NULL`?

Reply:
hmmm the 0xFFFF PanID means the node is not associated to any PAN coordinator. So, it's officially a valid configuration.

But I would be ok if `ext_addr` and `short_addr` can be NULL. After all, these are mostly one time operations, Changing the PanID is more recurrent (e.g when associating to a new PAN)

Reply:
Done. I also added the same note to the `set_csma_params` function for the exponential params.